code-styling/mysql/README.md

```diff
@@ -0,0 +1,574 @@
+- Keywords that consist of multiple words MUST be written on one line
+- Queries SHOULD be parametrized statements
+- Parameters SHOULD be specified by name
+- Operators MUST be surrounded by one space
+- Table names, column names and database names SHOULD NOT be surrounded with backticks
+
+### Select queries
+
+- If a `SELECT` query consists of more than one table, the most important one
+  SHOULD be used in the `FROM` clause
+
+- Table names and column names related to keywords MUST be on the same line,
+  except for the `SELECT` keyword (see below)
+
+- If a `SELECT` query selects one column, this column name MAY be written
+  on the same line as the `SELECT` keyword
+
+- If a `SELECT` query selects more than one column, the column name MUST be
+  written on separate lines with one indent level
+
+- If `AND` and `OR` keywords are used in a `WHERE` clause or `ON` clause,
+  brackets `()` are REQUIRED to specify precedence.
+
+Good select queries:
+
+```sql
+SELECT column_a
+FROM my_table
+WHERE id = 5;
+
+SELECT
+  column_a,
+  column_b,
+  column_c
+FROM my_table
+WHERE id < 100
+AND created_at >= '2017-01-01';
+
+SELECT
+  column_a,
+  column_b,
+  column_c
+FROM my_table
+WHERE (
+  id < 100
+  AND created_at >= '2017-01-01'
+) OR (
+  id < 50
+  AND created_at > '2016-06-30 12:34:56'
+)
+```
+
+Bad select queries:
+
+```sql
+SELECT column_a
+FROM
+  my_table
+WHERE id = 5;
+
+SELECT column_a
+FROM my_table
+WHERE
+  id = 5;
+
+SELECT column_a, column_b, column_c
+FROM my_table
+WHERE id < 100 AND created_at >= '2017-01-01';
+
+-- Missing brackets
+SELECT
+  column_a,
+  column_b,
+  column_c
+FROM my_table
+WHERE id < 100
+AND created_at >= '2017-01-01'
+OR id < 50
+AND created_at > '2016-06-30 12:34:56'
+```
+
+### Joins
+
+- The `JOIN` keyword MUST NOT be used without specifying the join type
+
+- The preferred join type is `INNER JOIN`
+
+- `LEFT JOIN` SHOULD only be used if the results may contain null values
+  for the joined table
+
+- `RIGHT JOIN` SHOULD be avoided
+
+- The `ON` clause MUST be written on the next line with one indent level
+
+Good join queries:
+
+```sql
+SELECT
+  t1.id,
+  t1.column_a,
+  t2.column_b
+FROM table1 AS t1
+INNER JOIN table2 AS t2
+  ON t1.id = t2.table1_id;
+
+SELECT
+  t1.id,
+  t1.column_a
+FROM table1 AS t1
+LEFT JOIN table2 AS t2
+  ON t1.id = t2.table1_id
+  AND t2.table1_id IS NULL;
+
+SELECT
+  t1.id,
+  t1.column_a
+FROM table1 AS t1
+LEFT JOIN table2 AS t2
+  ON t1.id = t2.table1_id
+  AND t2.updated_at IS NULL;
+```
+
+Bad join queries:
+
+```sql
+SELECT
+  t1.id,
+  t1.column_a,
+  t2.column_b
+FROM table1 AS t1
+INNER JOIN table2 AS t2 ON t1.id = t2.table1_id;
+
+SELECT
+  t1.id,
+  t1.column_a
+FROM table1 AS t1
+LEFT JOIN table2 AS t2
+ON t1.id = t2.table1_id
+AND t2.table1_id IS NULL;
+
+SELECT
+  t1.id,
+  t1.column_a
+FROM table1 AS t1
+LEFT JOIN table2 AS t2
+  ON t1.id = t2.table1_id AND t2.table1_id IS NULL;
+```
+
+### Delete queries
+
+Delete queries SHOULD not be used.
```

Comment:
If that is the argumentation, i would jus remove it, because having a way to rollback state should not be defined in a styleguide for SQL. An advise could be given, but that advise should not use the terms must or shall.